File: cafe-web/src/types/table.ts
import type Order from "./order";

export interface TableItem {
  id: number;
  createdAt: Date;
  name: string;
  capacity: number;
  locationX: number;
  locationZ: number;
  status: 'Available' | 'Occupied' | 'OutOfOrder';
  orders?: Order[];
}

export interface CreateTableDTO {
  name: string;
  capacity: number;
  locationX: number;
  locationZ: number;
}

export interface UpdateTableDTO {
  id: number;
  name?: string;
  capacity?: number;
  locationX?: number;
  locationZ?: number;
  status?: 'Available' | 'Occupied' | 'OutOfOrder';
}

File: cafe-api/Entities/Models/Table.cs
namespace Entities.Models
{
    public class Table
    {
        public int Id { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public String Name { get; set; } = null!;
        public int Capacity { get; set; }
        public decimal LocationX { get; set; }
        public decimal LocationZ { get; set; }
        public ICollection<Order>? Orders { get; set; }
        public TableStatus Status { get; set; } = TableStatus.Available;
    }

    public enum TableStatus
    {
        Occupied,
        Available,
        OutOfOrder
    }
}
